examples/text-classification/run_glue.py

```diff
@@ -350,11 +350,29 @@ def compute_metrics(p: EvalPrediction):
 
     # Training
     if training_args.do_train:
-        trainer.train(
+        train_result = trainer.train(
             model_path=model_args.model_name_or_path if os.path.isdir(model_args.model_name_or_path) else None
         )
+        metrics = train_result.metrics
+        metrics["train_n_objs"] = len(train_dataset)
```

Comment:
This is logged already at the beginning of training under `num_examples`, so not sure it adds anything.

Reply:
it doesn't log speed, or are you referring to just `train_n_objs`?

Ideally we should refactor all these scripts to use the same format and sequence, and helper functions